test/lib/differ_test.rb

```diff
@@ -0,0 +1,130 @@
+    end
+
+    it "should return correct number of characters deleted preceeding offset when last diff is a deletion" do
+      diffs = Differ.get_diffs("foo bar lorum ipsum", "foo bar ipsum")
+      assert_equal -3, Differ.get_delta_at_offset(diffs, 11)
+    end
+
+    it "should handle deletions immediately followed by insertions" do
+      diffs = Differ.get_diffs("foo bar lorum ipsum", "foo super lorum ipsum")
+      assert_equal 2, Differ.get_delta_at_offset(diffs, 11)
+    end
+
+    it "ignore changes that come after the supplied offset" do
+      diffs = Differ.get_diffs("foobar", "foo bar")
+      assert_equal 0, Differ.get_delta_at_offset(diffs, 3)
+    end
+  end
+
+  describe Differ, :range_was_deleted? do
+    it "should return true if the specified range was a subset of a deleted range of text" do
+      diffs = Differ.get_diffs("foo bar buzz", "foo buzz")
+      assert Differ.range_was_deleted? diffs, 5..6
+    end
+
+    it "should return true if the specified range was the exact range of a deleted range of text" do
+      diffs = Differ.get_diffs("foo bar buzz", "foo buzz")
+      assert Differ.range_was_deleted? diffs, 4..7
+    end
+
+    it "should return false if any part of the specified range was not part of a deleted range of text" do
+      diffs = Differ.get_diffs("foo bar buzz", "foo buzz")
+      refute Differ.range_was_deleted? diffs, 4..9
+    end
+  end
+
+  describe Differ, :adjust_offset do
+    it "should shift offset forward when text is prepended" do
+      before = "this is the text"
+      after = "and this is the text"
+      offset = 10
+      delta = after.length - before.length
+      diffs = Differ.get_diffs(before, after)
+      assert_equal offset + delta, Differ.adjust_offset(diffs, offset)
+    end
+
+    it "should shift offset backward when text is removed from beginning" do
+      before = "this is the text"
+      after = "is the text"
+      offset = 10
+      delta = after.length - before.length
+      diffs = Differ.get_diffs(before, after)
+      assert_equal offset + delta, Differ.adjust_offset(diffs, offset)
+    end
+
+    it "should shift offset backward when text is removed from end" do
+      before = "this is the text"
+      after = "this is the"
+      offset = 14
+      diffs = Differ.get_diffs(before, after)
+      assert_equal after.length, Differ.adjust_offset(diffs, offset)
+    end
+
+    it "should return identical offset when text is identical" do
+      text = "this is the text"
+      offset = 10
+      diffs = Differ.get_diffs(text, text)
+      assert_equal offset, Differ.adjust_offset(diffs, offset)
+    end
+
+    it "should not extend offset when text inserted after offset" do
+      diffs = Differ.get_diffs("foobar", "foo bar")
+      offset = 3
+      assert_equal offset, Differ.adjust_offset(diffs, offset)
+    end
+
+    it "should deal with spaces after word deletion" do
+      skip "we may or may not want this"
+    end
```

Comment:
Oh right. I was originally envisioning implementing something akin to a `trim()` so that if you had (note: brackets = annotation) `foo [bar bizz] lipsum` and you deleted `bizz ` (notice the trailing space) you'd validate that you ended up with `foo [bar] lipsum` rather than `foo [bar ]lipsum`. But I hesitate to get too smart with things.

Reply:
Aha! Gotcha. Yeah, leaving that feature out seems fine to me too, even more predictable!

Reply:
Removed.